### PR TITLE
add metrics for indexer and sidecar

### PR DESCRIPTION
**Pull request by rinx**
### Description:

Added new metrics for indexer and sidecar.

### Related Issue:

#343 

### How Has This Been Tested?:

nothing.

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [X] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/sidecar/service/observer/hook.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	StartTime time.Time
+	EndTime   time.Time
+	Bytes     int64
+
+	*storage.StorageInfo
+}
+
+type Hook interface {
+	BeforeProcess(ctx context.Context, info *BackupInfo) (context.Context, error)
+	AfterProcess(ctx context.Context, info *BackupInfo) error
```
Comment (kpango) [inline pkg/agent/sidecar/service/observer/hook.go:37]: How about naming as `PreProcess` & `PostProcess`

Comment (rinx) [reply pkg/agent/sidecar/service/observer/hook.go:37]: Both is okay, but i think it is better to use the current naming.
it is come from go-redis's Hook interface, it is already used in metrics/db/kvs.
https://github.com/vdaas/vald/blob/782e84b7a4109d4cf4b780d1f71c6a919074e573/internal/observability/metrics/db/kvs/redis/redis.go#L152

Also, the word "preprocess" sounds like modifying the incoming data. but here, not modifying just exporting metadata.

Comment (kpango) [reply pkg/agent/sidecar/service/observer/hook.go:37]: Both is okay too. But I don't think pre & post always modifying the data.
but I didn't know about go-redis naming, it's okay this time.

**Pull request merged**